gulpfile.js

```diff
@@ -0,0 +1,16 @@
+var gulp = require('gulp');
+var mocha = require('gulp-mocha');
+
+gulp.task('run-tests', function() {
+  gulp.src(['test/*.js'], { read: false })
+    .pipe(mocha({
+      ui: 'bdd'
+    }));
+});
+
+gulp.task('watch-files', function() {
+  gulp.watch(['./*.js', 'test/**/*.js'], ['run-tests']);
```

Comment:
fyi - technically this would watch your gulp file, but wouldn't actually restart the gulp process. for this simple assignment no biggy, but as you get into larger projects typically move js into some child folder.